app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -7,108 +7,95 @@ import { Section } from 'components/ucsf/Section/Section';
-    rawEligibilityData[resourceSlug],
+    resourceEligibilityGroups,
   );
 
-  // Todo: This setEligibilityGroup and toggleChecked logic could change pretty drastically
-  // once the API returns eligibility data. The shape of that data is still under discussion
-  // between product and dev
-  const setEligibilityGroup = (index: number, updatedEligibilityGroup: EligibilityGroup) => {
-    const updatedList = [
-      ...eligibilityGroupList.slice(0, index),
-      updatedEligibilityGroup,
-      ...eligibilityGroupList.slice(index + 1),
-    ];
-
-    setEligibilityGroupList(updatedList);
-  };
-
-  const toggleChecked = (eligibilityGroup: EligibilityGroup, eligibilityIndex: number) => {
-    const targetEligibility = eligibilityGroup.eligibilities[eligibilityIndex];
-    const targetToggleState = !targetEligibility.checked;
-    let updatedEligibilities;
-
-    // If a user toggles "See all", we (un)check all of the eligibilities accordingly
-    if (targetEligibility.name === 'See all') {
-      updatedEligibilities = massToggleGroupEligibilities(
-        eligibilityGroup.eligibilities,
-        targetToggleState,
-      );
+  const handleEligibilityClick = (
+    eligibility: Eligibility,
+    eligibilities: Eligibility[],
+  ) => {
+    const seeAllEligibilityItem = eligibilities.find(e => e.isSeeAll);
+    const eligibilityCheckedId = eligibility.checkedId;
+    const targetValue = !selectedEligibilities[eligibilityCheckedId];
+
+    if (eligibility.isSeeAll) {
+      // Check or uncheck all boxes in accordance with "See All" checked value
+      massToggleGroupEligibilities(eligibilities, targetValue);
     } else {
-      // If a user untoggles a specific eligiblity, this code also untoggles "See all"
-      // Currently, there is no logic to toggle "See all" if a user toggles all individual
-      // eligibilities given that this is somewhat of an implicit understanding; however, this
-      // can be added in the future
-      const seeAllEligibility = eligibilityGroup.eligibilities[0];
-      if (!targetToggleState) {
-        seeAllEligibility.checked = false;
-      }
-
-      const updatedEligibility = {
-        ...targetEligibility,
-        checked: !eligibilityGroup.eligibilities[eligibilityIndex].checked,
+      const updatedEligibilities = {
+        ...selectedEligibilities,
+        [eligibilityCheckedId]: targetValue,
       };
 
-      updatedEligibilities = [
-        { ...seeAllEligibility },
-        ...eligibilityGroup.eligibilities.slice(1, eligibilityIndex),
-        updatedEligibility,
-        ...eligibilityGroup.eligibilities.slice(eligibilityIndex + 1),
-      ];
-    }
+      // If target checked value is false, uncheck "See All" box as well
+      if (!targetValue) {
+        // N.B. Use of "!" assumes that every Eligibility array will have a "See All" element
+        updatedEligibilities[seeAllEligibilityItem!.checkedId] = false;
```

Comment:
I think it might be clearer to do the not-null assertion when you first assign the variable, since it would be right next to where you fetched the eligibility item in the first place and if in the future we wanted to use `seeAllEligibilityItem` in multiple places, it would also avoid having to do the not-null assertion at every place where you  use the variable.

Reply:
Good call. For now, since that var exists only within the fairly small scope of this function, I have just added a check in the `if` condition right above line 55 to ensure that the `seeAllEligibilityItem` exists before accessing the `checkedId` prop. I could see a case where it may not exist in some eligibility groups in the future.